relayer/connection-tx.go

```diff
@@ -324,3 +332,36 @@ func InitializeConnection(src, dst *Chain, srcUpdateHeader, dstUpdateHeader *tmc
 		return false, true, fmt.Errorf("connection ends already created")
 	}
 }
+
+// FindMatchingConnection will determine if there already exists a unintialized connection between source and counterparty.
+func FindMatchingConnection(source, counterparty *Chain) (string, bool) {
+	// TODO: add appropriate offset and limits, along with retries
+	connectionsResp, err := source.QueryConnections(0, 1000)
+	if err != nil {
+		if source.debug {
+			source.Log(fmt.Sprintf("Error: querying connections on %s failed: %v", source.PathEnd.ChainID, err))
+		}
+		return "", false
+	}
+
+	for _, connection := range connectionsResp.Connections {
+		if IsMatchingConnection(source, counterparty, connection) {
+			// unused connection found
+			return connection.Id, true
+		}
+	}
+
+	return "", false
+}
+
+// IsMatchingConnection determines if given connection matches required conditions
+func IsMatchingConnection(source, counterparty *Chain, connection *conntypes.IdentifiedConnection) bool {
+	// determines version we use is matching with given versions
+	_, isVersionMatched := conntypes.FindSupportedVersion(&conntypes.Version{}, conntypes.ProtoVersionsToExported(connection.Versions))
+	return connection.ClientId == source.PathEnd.ClientID &&
+		connection.Counterparty.ClientId == counterparty.PathEnd.ClientID &&
+		isVersionMatched && connection.DelayPeriod == defaultDelayPeriod &&
+		connection.Counterparty.Prefix.String() == defaultChainPrefix.String() &&
+		(connection.State == conntypes.UNINITIALIZED || connection.State == conntypes.OPEN &&
```

Comment:
```suggestion
		(connection.State == conntypes.OPEN &&
```
a connection set on chain should never be uninitialized